dali/pipeline/operators/detection/random_crop.cc

```diff
@@ -32,6 +32,7 @@ As an input, it accepts image, bounding boxes and labels. At the output
 cropped image, cropped and valid bounding boxes and valid labels are returned.)code")
   .NumInput(3)   // [img, bbox, label]
   .NumOutput(3)  // [img, bbox, label]
+  .DisallowInstanceGrouping()
```

Comment:
What about:
- hostdecoderrandomcrop
- nvJPEGDecoderRandomCrop
- SSDRandomCrop

Reply:
Also, why random_crop cannot be grouped?

Reply:
> What about:
> 
>     * hostdecoderrandomcrop
> 
>     * nvJPEGDecoderRandomCrop
> 
>     * SSDRandomCrop

First two (and random_crop) derive from RandomCropAttr which creates a batch of RandomCropGenerator. So it basically stores a Mersenne Twister per sample and uses it to generate crop windows.

As per SSDRandomCrop it cannot be used in this form as it uses one RNG in nondeterministic order. Same with RandomBBoxCrop.

Reply:
So SSDRandomCrop should be marked in this way as well.
BTW - why RandomBBoxCrop and SSDRandomCrop cannot have a batch of RNG?

Reply:
It is marked, you added a comment at that exact point. The file names can be confusing.
They can: #1109.